common.gypi

```diff
@@ -264,6 +264,16 @@
         }
       }
     },
+
+    # Defines these mostly for node-gyp to pickup, and warn addon authors of
+    # imminent V8 deprecations, also to sync how dependencies are configured.
+    'defines': [
+      # `OPENSSL_THREADS` is defined via GYP for openSSL for all architectures.
+      'OPENSSL_THREADS'
+      'V8_DEPRECATION_WARNINGS',
+      'V8_IMMINENT_DEPRECATION_WARNINGS',
```

Comment:
Just to understand this better, what are the differences between these two?

Reply:
Example: https://github.com/nodejs/node/pull/23414/files
AFAIU `V8_DEPRECATION_WARNINGS` makes API points marked with `V8_DEPRECATED` marked [`[[deprecated]]`](https://en.cppreference.com/w/cpp/language/attributes/deprecated)
`V8_IMMINENT_DEPRECATION_WARNINGS` makes thant for API points marked with `V8_DEPRECATE_SOON`.
Similar to node's policy `V8_DEPRECATED` are not-recommended for use, and can be removed in the next V8 version. `V8_DEPRECATE_SOON` can only be demoted to `V8_DEPRECATED` in a next version.

/CC @hashseed, right?

Reply:
ATM we do turn on `V8_DEPRECATION_WARNINGS` for node build, and native addons builds.

Reply:
So `V8_IMMINENT_DEPRECATION_WARNINGS` will report `DEPRECATE_SOON` and `V8_DEPRECATION_WARNINGS` will report the normal deprecation, right?